packages/netlify-cms-backend-gitlab/src/API.ts

```diff
@@ -0,0 +1,702 @@
+      // TODO: only send author params if they are defined.
+      unsentRequest.withParams(commitParams),
+      this.request,
+    ])(`${this.repoURL}/repository/files/${encodeURIComponent(path)}`);
+  };
+
+  generateContentKey(collectionName: string, slug: string) {
+    return generateContentKey(collectionName, slug);
+  }
+
+  contentKeyFromBranch(branch: string) {
+    return branch.substring(`${CMS_BRANCH_PREFIX}/`.length);
+  }
+
+  branchFromContentKey(contentKey: string) {
+    return `${CMS_BRANCH_PREFIX}/${contentKey}`;
+  }
+
+  async getMergeRequests(sourceBranch?: string) {
+    const mergeRequests: GitLabMergeRequest[] = await this.requestJSON({
+      url: `${this.repoURL}/merge_requests`,
+      params: {
+        state: 'opened',
+        labels: 'Any',
+        // eslint-disable-next-line @typescript-eslint/camelcase
+        target_branch: this.branch,
+        // eslint-disable-next-line @typescript-eslint/camelcase
+        ...(sourceBranch ? { source_branch: sourceBranch } : {}),
+      },
+    });
+
+    return mergeRequests.filter(
+      mr => mr.source_branch.startsWith(CMS_BRANCH_PREFIX) && mr.labels.some(isCMSLabel),
+    );
+  }
+
+  async listUnpublishedBranches() {
+    console.log(
+      '%c Checking for Unpublished entries',
+      'line-height: 30px;text-align: center;font-weight: bold',
+    );
+
+    const mergeRequests = await this.getMergeRequests();
+    const branches = mergeRequests.map(mr => mr.source_branch);
+
+    return branches;
+  }
+
+  async isFileExists(path: string, branch: string) {
+    const fileExists = await this.requestText({
+      method: 'HEAD',
+      url: `${this.repoURL}/repository/files/${encodeURIComponent(path)}`,
+      params: { ref: branch },
+      cache: 'no-store',
+    })
+      .then(() => true)
+      .catch(error => {
+        if (error instanceof APIError && error.status === 404) {
+          return false;
+        }
+        throw error;
+      });
+
+    return fileExists;
+  }
+
+  async getBranchMergeRequest(branch: string) {
+    const mergeRequests = await this.getMergeRequests(branch);
+    if (mergeRequests.length <= 0) {
+      throw new EditorialWorkflowError('content is not under editorial workflow', true);
+    }
+
+    return mergeRequests[0];
+  }
+
+  async getDifferences(to: string) {
+    const result: { diffs: GitLabCommitDiff[] } = await this.requestJSON({
+      url: `${this.repoURL}/repository/compare`,
+      params: {
+        from: this.branch,
+        to,
+      },
+    });
+
+    return result.diffs;
+  }
+
+  async retrieveMetadata(contentKey: string) {
+    const [collection, slug] = contentKey.split('/');
+    const branch = this.branchFromContentKey(contentKey);
+    const mergeRequest = await this.getBranchMergeRequest(branch);
+    const diff = await this.getDifferences(mergeRequest.sha);
+    const path = diff.find(d => d.old_path.includes(slug))?.old_path as string;
+    // TODO: get real file id
```

Comment:
The file id is used with Large Media support in git-gateway.
We should be able to get it using a`HEAD` request once https://gitlab.com/gitlab-org/gitlab/issues/194897 is resolved